procbuild/builder.py

```diff
@@ -77,99 +76,188 @@ def shell(cmd, path=None, retry=0):
+    return shell(f'git clone {repo} --branch {branch} --single-branch {build_path}', retry=4)
+
+    
+class BuildError(Exception):
+    
+    def __init__(self, message):
+        self.message = f"A build error occurred during the self.{message} step."
+
+        
+class BuildManager:
+    
+    def __init__(self, 
+                 user, 
+                 branch, 
+                 target, 
+                 master_branch='master', 
+                 log=None):
+        self.user = user
+        self.build_output = ''
+        self.status = 'fail'
+        self.branch = branch
+        self.target = target
+        self.master_branch = master_branch
+        self.build_status = 'Build started...'
+        self.build_pdf_path = ''
+        self.master_repo_path = joinp(cache(), 'scipy_proceedings')
+        self.build_timestamp = time.strftime('%d/%m %H:%M')
+        self.target_path = joinp(cache(), '%s.pdf' % target)
+        self.build_path = None
+        
+        data_filenames = ['IEEEtran.cls', 
+                          'draftwatermark.sty', 
+                          'everypage.sty']
+        self.data_files = [joinp(base_path, 'data', f) for f in data_filenames]
+
+    def add_output(self, msg):
+        self.build_output += msg
+
+    @property
+    def status_report(self):
+        return {'status': self.status,
+                'data': {'build_status': self.build_status,
+                         'build_output': self.build_output,
+                         'build_pdf_path': self.build_pdf_path,
+                         'build_timestamp': self.build_timestamp}
+                }
+
+
+    def _get_build_tools(self):
+        """This command will get the latest version of the repo path.
+        
+        """
+        if not os.path.exists(self.master_repo_path):
+            self.add_output('[*] Checking out proceedings build tools '
+                            f'to {self.master_repo_path}...\n')
+            errcode, output = checkout(repo('scipy-conference'), 
+                                       self.master_branch,
+                                       self.master_repo_path)
+        else:
+            self.add_output('[*] Updating proceedings build tools in'
+                            ' {self.master_repo_path}...\n')
+            errcode, output = shell('git pull', self.master_repo_path, retry=2)
+        
+        self.add_output(output)
+        
+        if errcode:
+            self.add_output('[X] Error code %d ' % errcode)
+            raise BuildError("get_build_tools")
+        
+        self._remove_papers_dir()
+        
+    
+    def _remove_papers_dir(self):
+        """ this command will remove the papers dir from the build_tools dir
+        
+        """
+        errcode, output = shell('rm -rf papers', 
+                                path=self.master_repo_path)
+        
+        self.add_output(output)
+
+        if errcode:
+            self.add_output('[X] Error code %d ' % errcode)
```

Comment:
`[X] Clean out tools papers directory` or something similar?

Reply:
Ok — I was trying to directly map it onto what was being done before as a simple fix.